song_rater/models/song.py

```diff
@@ -6,6 +6,11 @@ class Song(models.Model):
     title = models.CharField(max_length=100)
     artist = models.CharField(max_length=100)
 
+    class Meta:
+        constraints = [
+            models.UniqueConstraint(fields=['title', 'artist'], name='unique_song')
+        ]
+
```

Comment:
Do you want to add a test for this constraint? Just to make sure you handle a failure of the restraint in the intended way? (What is the intended way? A nice message to the user or a generic non-201 error page?)

Reply:
I added this constraint test. Django should generate an error message to the user when attempting to add an already existing song. So now I test that that is included in the response.

Reply:
I stand corrected. I check for the 302 status code, which only happens when a duplicate song is added